pobieracz_danych_gugik.py

```diff
@@ -1696,15 +1671,45 @@ def filterAerotriangulacjaListAndRunTask(self, aerotriangulacjaList):
                 QgsApplication.taskManager().addTask(task)
                 QgsMessageLog.logMessage('runtask')
 
+    def filterMeshListAndRunTask(self, mesh_objs):
+        mesh_objs = list(set(mesh_objs))
+        if not mesh_objs:
+            msgbox = QMessageBox(
+                QMessageBox.Information,
+                'Komunikat',
+                'Nie znaleniono danych spełniających kryteria'
+            )
+            msgbox.exec_()
+            return
+        msgbox = QMessageBox(
+            QMessageBox.Question,
+            'Potwierdź pobieranie',
+            f'Znaleziono {len(mesh_objs)} plików spełniających kryteria. Czy chcesz je wszystkie pobrać?'
+        )
+        msgbox.addButton(QMessageBox.Yes)
+        msgbox.addButton(QMessageBox.No)
+        msgbox.setDefaultButton(QMessageBox.No)
+        if msgbox.exec() == QMessageBox.No:
+            return
+        task = DownloadMesh3dTask(
+            description='Pobieranie plików danych siatkowych modeli 3D',
+            mesh_objs=mesh_objs,
+            folder=self.dockwidget.folder_fileWidget.filePath(),
+            iface=self.iface
```

Comment:
Podobnie, proszę utwórz zmienną self.output_folder, która będzie odczytywała folder raz dla wszystkich innych funkcji (chyba najlepiej w funkcji run)

Reply:
będzie na to osobny PR